lib/internal/test_runner/harness.js

```diff
@@ -272,6 +272,7 @@ function lazyBootstrapRoot() {
     };
     const globalOptions = parseCommandLine();
     globalOptions.cwd = process.cwd();
+    globalOptions.timeout = Infinity;
```

Comment:
Since this is unconditionally being set to `Infinity` wouldn't it make more sense to not make the change on line 274 of `lib/internal/test_runner/utils.js`, which corresponds to this code path?